lib/pf/provisioner/intune.pm

```diff
@@ -233,6 +232,29 @@ sub get_device_info {
     return $self->decode_response($curl_info, $response_body);
 }
 
+sub get_device_info {
+    my ($self, $mac) = @_;
+    my $logger = get_logger();
+
+    my @infos;
+
+    my $info = $self->perform_get_device_info($mac, $self->protocol.'://' . $self->host . ':' .  $self->port . '/v1.0/deviceManagement/managedDevices?$select=wiFiMacAddress,complianceState,id');
+    if($info == $pf::provisioner::COMMUNICATION_FAILED) {
+        return $pf::provisioner::COMMUNICATION_FAILED;
+    }
+    push @infos, $info;
+
+    while($info && $info != $pf::provisioner::COMMUNICATION_FAILED && $info->{'@odata.nextLink'}) {
+        $info = $self->perform_get_device_info($mac, $info->{'@odata.nextLink'});
+        if($info == $pf::provisioner::COMMUNICATION_FAILED) {
+            return $pf::provisioner::COMMUNICATION_FAILED;
```

Comment:
If the previous messages succeeded shouldn't you just stop and return what was received prior instead of throw away all the previous messages?

Reply:
Yes we do throw it away so that it can be retried from scratch. Otherwise you could have an incomplete list if we would return what we already have